babel/messages/catalog.py

```diff
@@ -142,6 +142,21 @@ def __eq__(self, other):
     def __ne__(self, other):
         return self.__cmp__(other) != 0
 
+    def is_eq(self, other):
+        def values_to_compare(obj):
+            return (
+                obj.id,
+                obj.string,
+                obj.locations,
+                obj.flags,
+                obj.auto_comments,
+                obj.user_comments,
+                obj.previous_id,
+                obj.lineno,
+                obj.context,
+            )
+        return values_to_compare(self) == values_to_compare(other)
```

Comment:
I was a bit hesitant about using `__dict__`, in case there would be any private properties or such that would cause problems, but looking at it now again it looks like we should be good.